src/app/3d/qgsmaptoolclippingplanes.cpp

```diff
@@ -112,40 +130,50 @@ void QgsMapToolClippingPlanes::canvasReleaseEvent( QgsMapMouseEvent *e )
+        crossSectionPolygon.set( nullptr );
+        QgsDebugError( QStringLiteral( "Could not reproject cross-section extent to 3d map canvas crs." ) );
+      }
+
       if ( !crossSectionPolygon.intersects( m3DCanvas->mapCanvas3D()->scene()->sceneExtent() ) )
       {
         clear();
         QgsMessageBar *msgBar = QgisApp::instance()->messageBar();
-        msgBar->pushInfo( QString(), tr( "The cross section is outside of the scene extent, please select a new one!" ) );
+        if ( crossSectionPolygon.isNull() )
+          msgBar->pushWarning( QString(), tr( "Could not reproject the cross-section extent to 3D map coordinates." ) );
+        else
+          msgBar->pushInfo( QString(), tr( "The cross section is outside of the scene extent, please select a new one!" ) );
       }
       else
       {
-        const QgsVector3D startPoint = QgsVector3D( mRubberBandPoints->getPoint( 0, 0 )->x(), mRubberBandPoints->getPoint( 0, 0 )->y(), 0 );
-        const QgsVector3D endPoint = QgsVector3D( mRubberBandPoints->getPoint( 0, 1 )->x(), mRubberBandPoints->getPoint( 0, 1 )->y(), 0 );
-        const QList<QVector4D> clippingPlanes = Qgs3DUtils::lineSegmentToClippingPlanes(
-          QgsVector3D( startPoint.x(), startPoint.y(), 0 ),
-          QgsVector3D( endPoint.x(), endPoint.y(), 0 ),
+        QgsPointXY pt0 = *mRubberBandPoints->getPoint( 0, 0 );
+        QgsPointXY pt1 = *mRubberBandPoints->getPoint( 0, 1 );
+        try
+        {
+          pt0 = mCt->transform( pt0 );
+          pt1 = mCt->transform( pt1 );
+        }
+        catch ( const QgsCsException & )
+        {
+          QgsMessageBar *msgBar = QgisApp::instance()->messageBar();
+          msgBar->pushWarning( QString(), tr( "Could not reproject the cross-section extent to 3D map coordinates." ) );
+        }
+
+        m3DCanvas->mapCanvas3D()->enableCrossSection(
+          pt0,
+          pt1,
           mRectangleWidth,
-          m3DCanvas->mapCanvas3D()->mapSettings()->origin()
-        );
-
-        // calculate the middle of the front side defined by clipping planes
-        QgsVector linePerpVec( ( endPoint - startPoint ).x(), ( endPoint - startPoint ).y() );
-        linePerpVec = -linePerpVec.normalized().perpVector();
-        const QgsVector3D linePerpVec3D( linePerpVec.x(), linePerpVec.y(), 0 );
-        const QgsVector3D frontStartPoint( startPoint + linePerpVec3D * mRectangleWidth );
-        const QgsVector3D frontEndPoint( endPoint + linePerpVec3D * mRectangleWidth );
-
-        const QgsCameraPose camPose = Qgs3DUtils::lineSegmentToCameraPose(
-          frontStartPoint,
-          frontEndPoint,
-          m3DCanvas->mapCanvas3D()->scene()->elevationRange( true ),
-          m3DCanvas->mapCanvas3D()->scene()->cameraController()->camera()->fieldOfView(),
-          m3DCanvas->mapCanvas3D()->mapSettings()->origin()
+          true
         );
 
-        m3DCanvas->enableClippingPlanes( clippingPlanes, camPose );
+        m3DCanvas->crossSectionToolFinished();
```

Comment:
this tight binding between the classes is a bit ugly -- it'd be nicer to emit a signal here, and make the connection to crossSectionToolFinished in the 3d canvas alone (and make crossSectionToolFinished a private slot)